chattoreagent/src/main/kotlin/ChattoreAgent.kt

```diff
@@ -28,12 +33,53 @@ class ChatListener : Listener {
     }
 }
 
+class AfkListener(private val afkMap: ConcurrentHashMap<UUID, Long>) : Listener {
```

Comment:
It could be worth it to start on feature organization here too, like putting everything afk-related into AFK.kt and have a function there to do the enabling stuff. The placeholder expansion will be a bit derpy... it could live with the AFK code for as long as there is only AFK-related placeholders there though.